extensions/amp-script/amp-script.md

```diff
@@ -77,63 +109,103 @@ You can also include your JavaScript inline, in a `script` tag. You must:
 For security reasons, `amp-script` elements with a `script` or cross-origin `src` attribute require a [script hash](#script-hash) in a `<meta name="amp-script-src" content="...">` tag. Also, same-origin `src` files must have [`Content-Type`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Type): `application/javascript` or `text/javascript`.
 [/tip]
 
-## How does it work?
+### How does it work?
 
-`amp-script` runs your custom JavaScript in a [Web Worker](https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API/Using_web_workers) that has access to a virtual DOM. When your JavaScript code modifies this virtual DOM, `amp-script` forwards these changes to the main thread and applies them to the `amp-script` element subtree.
+`amp-script` runs your custom JavaScript in a [Web Worker](https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API/Using_web_workers) that has access to a virtual DOM. When the real DOM changes, `amp-script` updates the virtual DOM to match it. Similarly, when your JavaScript modifies the virtual DOM, `amp-script` updates the real DOM. In this way, the real DOM and virtual DOM are kept in sync.
```

Comment:
> When the real DOM changes, `amp-script` updates the virtual DOM to match it. 

This actually isn't the case today. The updates are only one way (virtual -> real DOM). This is part of the reason why many AMP components aren't supported -- AMP components modify the real DOM which makes diverge from the virtual DOM.

Reply:
Oh!

Now that you point this out, it makes utter sense. No AMP component is going to modify the children of the `<amp-script>`. (Though I've never tried embedding a binding like `<p [text]="foo">` inside an `<amp-script>`.) Unfortunately, I think all of our presentations have claimed that the updates are bidirectional 🙄

Reply:
Changed documentation. And warned users about letting the virtual and real DOM get out of sync in this way: https://github.com/ampproject/amphtml/issues/29917